packages/bech32m/src/format/convert.ts

```diff
@@ -16,7 +16,7 @@ export const fromBech32m = <P extends Prefix>(
   prefix: Prefix = b32mStr.slice(0, b32mStr.lastIndexOf('1')) as P,
 ): Uint8Array => {
   const p = b32mStr.slice(0, b32mStr.lastIndexOf('1'));
-  if (p !== prefix) throw new TypeError('Unexpected prefix');
+  if (p !== prefix) throw new TypeError(`Unexpected prefix: expected ${prefix}, got ${p}`);
```

Comment:
Ran into these errors when using these helpers with test data, so I added some detail to the errors to make them easier to parse.